bookkeeper-stats-providers/prometheus-metrics-provider/src/main/java/org/apache/bookkeeper/stats/prometheus/PrometheusMetricsProvider.java

```diff
@@ -172,6 +172,7 @@ public StatsLogger getStatsLogger(String scope) {
 
     @Override
     public void writeAllMetrics(Writer writer) throws IOException {
+        PrometheusTextFormatUtil.reset();
```

Comment:
This is a sign of something going wrong.

What happens with two concurrent calls to this method?

Reply:
@eolivelli you are right, we don't know what will happen. Let me push a fix.